src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/FileUploader.vue

```diff
@@ -64,16 +87,53 @@ export default defineComponent({
       selectedFiles: [],
       showErrorBanner: false,
       errorBannerMessage: "",
+      firstLoad: true,
     };
   },
+  computed: {
+    fileErrors() {
+      return this.selectedFiles.some((file) => file.fileErrors.length !== 0);
+    },
+    filesInProgress() {
+      return this.selectedFiles.some((file) => file.progress < 101);
+    },
+    fileTooLarge() {
+      return this.selectedFiles.some((file) => file.file.size > this.maxFileSizeInBytes);
+    },
+    allFilesTooLarge() {
+      const totalSize = this.selectedFiles?.reduce((acc, sf) => acc + sf.file.size, 0);
+      if (totalSize > this.maxFileSizeInBytes) {
+        return true;
+      }
+      return totalSize > this.maxFileSizeInBytes;
+    },
+    duplicateFileName() {
+      const fileNameMap = new Set();
+      for (const file of this.selectedFiles) {
+        if (fileNameMap.has(file.file.name)) {
+          return true; //there is a duplicate stop checking and return true
+        }
+        fileNameMap.add(file.file.name);
+      }
+
+      return false;
+    },
+  },
   watch: {
     selectedFiles: {
       handler() {
-        this.updateEmit(); // Emit updates whenever selectedFiles changes
+        if (!this.firstLoad) {
+          this.updateEmit(); // Emit updates whenever selectedFiles changes
+        } else {
+          this.firstLoad = false;
+        }
       },
       deep: true,
     },
   },
+  mounted() {
+    this.selectedFiles = [...this.userFiles];
```

Comment:
We should probably make `selectedFiles` a prop since we are needing to do this

Reply:
will address in another ticket to fix.